packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart

```diff
@@ -127,6 +143,41 @@ class _InstanceColumn extends ColumnData<DiffClassStats> {
 
   @override
   bool get numeric => true;
+
+  @override
+  Widget? build(
+    BuildContext context,
+    DiffClassStats data, {
+    bool isRowSelected = false,
+    VoidCallback? onPressed,
+  }) {
+    if (!FeatureFlags.evalAndBrowse) return null;
+    final objects = _instances(data);
+    final theHeap = heap;
+
+    // There is no detailed information, if [dataPart] equals to [_DataPart.delta].
+    if (objects is! ObjectSet || theHeap == null) return null;
```

Comment:
why not check if `dataPart == _DataPart.delta` then if that is the case we are looking for?

Reply:
Because I need type promotion. Added asserts.

Reply:
what about using `dataPart == _DataPart.delta` for the if statement since that is what we we are truly checking for? and then using local variables to get the type promotion you need. If `objects is! ObjectSet || theHeap == null` eventually becomes true for another `_DataPart` type, then this could lead to bugs where the if passes when we wouldn't expect it to